### PR TITLE
cpu: added stm32_common

### PR DESCRIPTION
**Pull request by haukepetersen**
in the same style as for the `nrf5x` and `sam21` here also a start to unify the `stm32` implementations and to get rid of some duplicated code.

### CODE REVIEW

File: cpu/stm32l1/ldscripts/stm32l151rc.ld [hunk 1]
```diff
@@ -22,15 +22,9 @@ MEMORY
...
-    /* see STM32L1 Reference Manual (31.2 Unique device ID registers (96 bits))
-     *    Base address:
-     *      - 0x1FF80050 for Cat.1 and Cat.2 devices
-     *      - 0x1FF800D0 for Cat.3, Cat.4, Cat.5 and Cat.6 devices
-     */
     cpuid (r)       : ORIGIN = 0x1ff800d0, LENGTH = 12
 }
 
-_cpuid_address = ORIGIN(cpuid);
+cpuid_addr = ORIGIN(cpuid);
```
Comment (DipSwitch) [inline cpu/stm32l1/ldscripts/stm32l151rc.ld]: _cpuid_address ?

Comment (haukepetersen) [inline cpu/stm32l1/ldscripts/stm32l151rc.ld]: ups, rebasing error... will fix

**Comment by haukepetersen** [comment]
fixed

**Comment by haukepetersen** [comment]
fixed little error in the cpuid.c and squashed it right in (accidentally...)

**Comment by cgundogan** [comment]
ACK. Tested with: iotlab-m3, several nucleo boards and the stm32f4discovery

**Comment by haukepetersen** [comment]
Travis failed on some tests, but they run fine locally. Rebased once more to give Travis another chance.

**Comment by haukepetersen** [comment]
Travis fails unrelated -> so go

**Comment by OlegHahm** [comment]
@cgundogan, I wonder how you tested. https://github.com/haukepetersen/RIOT/commit/ea8db10524c846a846c0750f9ba2b2e121baf6bb is causing a hardfault on iotlab-m3.

**Comment by haukepetersen** [comment]
This is really strange. I did fix this CPUID issue, just checked my local branch which should be identical to the PR branch, there this issue is actually fixed... Weird.

**Pull request merged**